pages/api/socketio-init.ts

```diff
@@ -0,0 +1,58 @@
+import axios from "axios"
+import { BehaviorSubject, defer, from, iif, timer } from "rxjs"
+import { map, switchMapTo } from "rxjs/operators"
+import { Server } from "socket.io"
+
+const { IEX_API_TOKEN: TOKEN } = process.env
+
+const sub = new BehaviorSubject<string>("")
+
+const ioHandler = (_req: any, res: any) => {
+  if (!res.socket.server.io) {
```

Comment:
Instead having logic nested in a long conditional,  test for the other case first and return early.

Reply:
👍